app/models/user.rb

```diff
@@ -581,39 +615,18 @@ def bust_cache
 
   def core_profile_details_changed?
     saved_change_to_username? ||
-      saved_change_to_name? ||
-      saved_change_to_summary? ||
-      saved_change_to_bg_color_hex? ||
-      saved_change_to_text_color_hex? ||
       saved_change_to_profile_image? ||
+      profile&.changed? || # TODO: @citizen428 This is not ideal, need better solution
```

Comment:
I want to call out that this may probably needs to be monitored as it could cause too many profile resaves. The alternative is to completely remove this for now and take some time to figure out a better way to do this. WDYT @mstruve?

Reply:
You are correct, we only need to do this for a few fields, which is why my comment calls out this code as less than ideal. 

The problem is that `core_profile_details_changed?` on `Profile` doesn't necessarily work. To make it reliable it would need to check `data_was` during the profile update, which may already have happened at this point.

But typing out this answer made me realize that this whole method should move into the `Profiles::Update` service object because that's the one place where I have all the necessary info to make this decision in the same place and it's also the canonical way to update users and profiles going forward.

Excelltent rubberducking, thanks!



I was thinking about this yesterday, the problem is that this